### ISSUE TITLE
[orc8r][1.4] Prometheus Pod crashlooping after upgrade

### ISSUE THREAD

**Issue by sudhikan** [magma/magma#4580]
### Your Environment

- **Version:** 1.4
- **Affected Component:** orc8r
- **Affected Subcomponent:** orc8r-prometheus
- **Deployment Environment:** AWS (orc8r)

### Describe the Issue
Upgraded from 1.3 to 1.4 and when `terraform apply` was running, noticed the orc8r-prometheus was crashlooping and preventing the `apply` from finishing. Deleting the old prom pod and the new one that was being created, fixed the issue. 

kubectl -n orc8r describe pod <orc8r-prometheus-pod-name> showed "Back-off restarting failed container". 

Below are the logs I collected for the pod before deleting:

```
kubectl -n orc8r logs -f orc8r-prometheus-7fc468d458-5gngc
level=info ts=2021-01-24T00:53:46.528Z caller=main.go:343 msg="Starting Prometheus" version="(version=2.20.1, branch=HEAD, revision=983ebb4a513302315a8117932ab832815f85e3d2)"
level=info ts=2021-01-24T00:53:46.529Z caller=main.go:344 build_context="(go=go1.14.6, user=root@7cbd4d1c15e0, date=20200805-17:26:58)"
level=info ts=2021-01-24T00:53:46.529Z caller=main.go:345 host_details="(Linux 4.14.209-160.339.amzn2.x86_64 #1 SMP Wed Dec 16 22:44:04 UTC 2020 x86_64 orc8r-prometheus-7fc468d458-5gngc (none))"
level=info ts=2021-01-24T00:53:46.529Z caller=main.go:346 fd_limits="(soft=1048576, hard=1048576)"
level=info ts=2021-01-24T00:53:46.529Z caller=main.go:347 vm_limits="(soft=unlimited, hard=unlimited)"
level=info ts=2021-01-24T00:53:46.574Z caller=main.go:684 msg="Starting TSDB ..."
level=info ts=2021-01-24T00:53:46.574Z caller=web.go:524 component=web msg="Start listening for connections" address=0.0.0.0:9090
level=info ts=2021-01-24T00:53:46.585Z caller=repair.go:59 component=tsdb msg="Found healthy block" mint=1611173387498 maxt=1611187200000 ulid=01EWHN0NBE2ZRCZFPGN1MWVTJT
level=info ts=2021-01-24T00:53:46.588Z caller=repair.go:59 component=tsdb msg="Found healthy block" mint=1611187200000 maxt=1611381600000 ulid=01EWQ7HKNQMJZEYZQY0FDH5HCH
level=info ts=2021-01-24T00:53:46.591Z caller=repair.go:59 component=tsdb msg="Found healthy block" mint=1611381600000 maxt=1611403200000 ulid=01EWQW4QHF1M0ACKTTN799STTT
level=info ts=2021-01-24T00:53:46.594Z caller=repair.go:59 component=tsdb msg="Found healthy block" mint=1611424800000 maxt=1611432000000 ulid=01EWRGQWJM6BC5RC13MJPSC156
level=info ts=2021-01-24T00:53:46.596Z caller=repair.go:59 component=tsdb msg="Found healthy block" mint=1611403200000 maxt=1611424800000 ulid=01EWRGQXE7R52B2Q7RG82X47CY
level=info ts=2021-01-24T00:53:46.599Z caller=repair.go:59 component=tsdb msg="Found healthy block" mint=1611432000000 maxt=1611439200000 ulid=01EWRQKKV00STDBXDHTX7YPZP7
level=info ts=2021-01-24T00:53:46.623Z caller=main.go:553 msg="Stopping scrape discovery manager..."
level=info ts=2021-01-24T00:53:46.623Z caller=main.go:567 msg="Stopping notify discovery manager..."
level=info ts=2021-01-24T00:53:46.623Z caller=main.go:589 msg="Stopping scrape manager..."
level=info ts=2021-01-24T00:53:46.623Z caller=main.go:549 msg="Scrape discovery manager stopped"
level=info ts=2021-01-24T00:53:46.623Z caller=main.go:563 msg="Notify discovery manager stopped"
level=info ts=2021-01-24T00:53:46.623Z caller=manager.go:888 component="rule manager" msg="Stopping rule manager..."
level=info ts=2021-01-24T00:53:46.623Z caller=manager.go:898 component="rule manager" msg="Rule manager stopped"
level=info ts=2021-01-24T00:53:46.623Z caller=notifier.go:601 component=notifier msg="Stopping notification manager..."
level=info ts=2021-01-24T00:53:46.623Z caller=main.go:755 msg="Notifier manager stopped"
level=info ts=2021-01-24T00:53:46.623Z caller=main.go:583 msg="Scrape manager stopped"
level=error ts=2021-01-24T00:53:46.623Z caller=main.go:764 err="opening storage failed: lock DB directory: resource temporarily unavailable"

```

**Comment by karthiksubraveti** [comment]
@sudhikan do you know if it was in a crashloop earlier ?

**Comment by sudhikan** [comment]
@karthiksubraveti it was not. The old pod was in "running" state.

**Comment by Scott8440** [comment]
tl;dr: prometheus pod has a lock on a db directory, and it k8s won't delete an old pod until a new one is already up, but the new one can't come up while the old one exists since it has a lock on the directory.

**Comment by Scott8440** [comment]
This can be resolved by manually deleting the terminating prometheus pod. k8s will then be able to spin up the new one. Nothing of value will be lost, since scrapes will resume immediately once the new pod comes up. 

I'll make an item for myself to look into how we can deploy prometheus in a way to prevent this in the future, as it will happen each time we upgrade the prometheus version. I believe we might be able to do it by deploying as a StatefulSet but I'm not sure yet

**Comment by hcgatewood** [comment]
@Scott8440 let's try to prioritize this. We need to get to a place where upgrades happen seamlessly, and manually deleting a pod is way too hands-on for partners
